src/main/java/hudson/remoting/Engine.java

```diff
@@ -145,6 +170,19 @@ public void setNoReconnect(boolean noReconnect) {
         this.noReconnect = noReconnect;
     }
 
+    public void setCandidateCertificates(List<X509Certificate> candidateCertificates) {
+        this.candidateCertificates = candidateCertificates == null
+                ? null
+                : new ArrayList<X509Certificate>(candidateCertificates);
+    }
+
+    public void addCandidateCertificate(X509Certificate certificate) {
```

Comment:
I added a `@NotThreadSafe` annotation... if you want to make it thread safe, that is a different change